python/paddle/fluid/dygraph/static_runner.py

```diff
@@ -0,0 +1,637 @@
+        self._program_desc = self._append_backward_desc()
+
+        # Step 5. recheck parameters stop gradients
+        self._recheck_stop_gradients()
+
+        # For Debug
+        # DescParser.print_program_desc(self._program_desc)
+
+    def train(self):
+        framework._dygraph_tracer().train_mode()
+        self._change_is_test_status(False)
+
+    def eval(self):
+        framework._dygraph_tracer().eval_mode()
+        self._change_is_test_status(True)
+
+    def forward(self, inputs):
+        # Step 1. check inputs, may affect performance
+        if not isinstance(inputs, (list, tuple)):
+            inputs = [inputs]
+
+        input_vars = []
+        for i, value in enumerate(inputs):
+            if not isinstance(value, (np.ndarray, core.VarBase)):
+                raise TypeError(
+                    "The type of inputs.value in StaticModelRunner.forward must be numpy array or Variable(VarBase), but received %s."
+                    % type(value))
+            # NOTE: In order to unify the API, firstly convert the input to VarBase
+            if isinstance(value, np.ndarray):
+                var = core.VarBase(
+                    value=value,
+                    name=self._input_names[i],
+                    persistable=False,
+                    place=framework._current_expected_place(),
+                    zero_copy=True)
+            else:
+                var = value
+                var.name = self._input_names[i]
+            # NOTE: after PR22939 [Add double grad], the grad op maker's
+            # SetOutput will set to None if the input var stop_gradient=True 
+            var.stop_gradient = False
+            input_vars.append(var)
+
+        params = []
+        for param in self._parameters.values():
+            params.append(param)
+
+        output_vars = []
+        for var_desc in self._output_descs:
+            var = core.VarBase(var_desc.dtype(),
+                               var_desc.shape(),
+                               var_desc.name(), var_desc.type(), False)
+            var.stop_gradient = False
+            output_vars.append(var)
+
+        # hold forward variables
+        out_scope = core.VarBase(core.VarDesc.VarType.FP32, [],
+                                 "program_out_scope",
+                                 core.VarDesc.VarType.STEP_SCOPES, True)
+
+        # Step 2. run prorgam by op
+        framework._dygraph_tracer().trace_op(
+            type='run_program',
+            inputs={'X': input_vars,
+                    'Params': params},
+            outputs={'Out': output_vars,
+                     'OutScope': out_scope},
+            attrs={
+                'global_block': self._program_desc.block(0),
+                'start_op_index': 0,
+                'end_op_index': self._load_program_desc.block(0).op_size()
+            })
+
+        outs = output_vars
+        if len(output_vars) == 1:
+            outs = output_vars[0]
```

Comment:
Does it necessary ?

Reply:
not, but keep consistent with inputs form, input or return list is alos be ok